core-services/src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -31,37 +31,27 @@ public HashPartitionStrategy(final HashPartitionStrategyCrutch hashPartitioningC
     }
 
     @Override
-    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+    public List<String> extractEventKeys(final EventType eventType, final JSONObject event)
```

Comment:
basically, the part which gets values from event by partitions key from metadata is not separate function? nice